### ISSUE TITLE
A question about the gradients of sampling

### ISSUE THREAD

**Issue by leoozy** [VDIGPKU/DADA#21]
Hello, thank you for your great works. I have a question about how you update the sampling parameters (arch parameters). You update the the sampling parameters with the validation set. But your do not augment the validation set with the sampled augmentation. That means that, the gradients of the loss respect to the validation data is None. Then how do you update the sampling parameters by the validation set?

**Comment by latstars** [comment]
The gradient is estimated by the bi-level optimization algorithm. You can find it on page 9 of [DADA](https://arxiv.org/pdf/2003.03780.pdf).
Actually, the gradient of augment parameters is calculated by the second-order gradient of $$L_{train}(w - d L_{val}(w), alpha)$$, which is not the same as gradient w.r.t. augment parameters on validation data.
